account_avatax_exemption_base/models/exemption.py

```diff
@@ -0,0 +1,179 @@
+from datetime import timedelta
+
+from odoo import _, api, fields, models
+
+
+class ResPartnerGroupState(models.Model):
+    _name = "res.partner.group.state"
+    _description = "Avatax Group of States"
+
+    name = fields.Char()
+    country_id = fields.Many2one("res.country")
+    state_ids = fields.Many2many(
+        "res.country.state",
+    )
+
+
+class ResPartnerExemptionLine(models.Model):
+    _name = "res.partner.exemption.line"
+    _description = "Avatax Exemption Line"
+
+    name = fields.Char(index=True, default=lambda self: _("New"))
+    exemption_id = fields.Many2one("res.partner.exemption", required=True)
+    partner_id = fields.Many2one(related="exemption_id.partner_id", store=True)
+    state_id = fields.Many2one("res.country.state")
+    avatax_id = fields.Char("Avatax Certificate ID", copy=False, readonly=True)
+    avatax_status = fields.Boolean(default=True)
+    linked_to_customer = fields.Boolean(readonly=True)
+    add_exemption_number = fields.Boolean()
+    exemption_number = fields.Char()
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get("name", _("New")) == _("New"):
+                vals["name"] = self.env["ir.sequence"].next_by_code(
+                    "exemption.line.sequence"
+                ) or _("New")
+        return super().create(vals_list)
+
+
+class ResPartnerExemptionBusinessType(models.Model):
+    _name = "res.partner.exemption.business.type"
+    _description = "Exemption Activity Type"
+
+    name = fields.Char(required=True)
+    avatax_id = fields.Char(required=True, readonly=True)
+
+
+class ResPartnerExemptionType(models.Model):
+    _name = "res.partner.exemption.type"
+    _description = "Avatax Exemption Type"
+
+    name = fields.Char()
+    business_type = fields.Many2one(
+        "res.partner.exemption.business.type",
+        string="Activity Type",
+    )
+    group_of_state = fields.Many2one(
+        "res.partner.group.state", string="Group of States"
+    )
+    state_ids = fields.Many2many("res.country.state", string="States")
+    exemption_validity_duration = fields.Integer(
+        help="Validity duration in days", default=30
+    )
+
+    @api.onchange("group_of_state")
+    def onchange_group_of_state(self):
+        if self.group_of_state.state_ids and not self.state_ids:
+            self.state_ids = [(6, 0, self.group_of_state.state_ids.ids)]
```

Comment:
```suggestion
            self.state_ids = [Command.set(self.group_of_state.state_ids.ids)]
```
using `Command` is IMHO better, see: https://www.odoo.com/documentation/17.0/developer/reference/backend/orm.html#odoo.fields.Command